tg-x/network-architecture.md

```diff
@@ -0,0 +1,118 @@
+where each node runs a set of processes, or actors.
+We call these *engines* in our design.
+Engines communicate with each other via message passing.
+
+Each node has a cryptographic identity that uniquely identifies it in the network.
+A node consists of a collection of running engines
+that send and respond to messages.
+Local engines communicate with each other directly,
+while they reach remote nodes via the Router and Transport engines.
+
+The Transport engine is responsible for maintaining authenticated and encrypted network connections between nodes
+via a modular transport system that supports various transport protocols with different reliability, latency, and privacy guarantees.
+Transport protocols initially supported include QUIC and TLS,
+later this can be extended with onion-routed, mix, mesh, and delay-tolerant network protocols.
+
+The Router engine routes incoming and outgoing messages from and to local engines,
+based on the destination address of each message, which consists of a node and engine identity.
+
+## Communication patterns
+
+The network supports unicast, multicast, and anycast communication patterns.
+
+- *Unicast*: direct messages between two engines.
+- *Multicast*: few-to-many messages sent to a publish-subscribe topic and delivered to all subscribers.
+- *Anycast*: messages sent to any known member of a domain.
+
+## Network architecture
+
+The network consists of nodes that may communicate with each other directly,
+and participate in any number of domains.
+
+Nodes in the core network are directly reachable via transport address(es)
+specified in their signed node advertisements.
+
+Mobile nodes of end-users may communicate directly on edge networks,
+and reach the core network via relays
+that store and forward messages for them.
+
+This allows reliable message delivery and basic location privacy for mobile nodes.
+Mobile nodes with higher location privacy requirements
+may use an onion-routed or mix network to connect to the core network.
+
+## Domains
+
+The network consists of sovereign domains that enable pluralistic interoperability.
+
+A domain is a sub-network with its own *cryptographic identity*.
+Domain creation is permissionless, it only requires a domain advertisement signed by the domain owner(s),
+which contains the domain configuration.
+Domain identity is based on a public key, and may use threshold cryptography.
+The domain configuration specifies the authentication mechanism used to join the domain,
+the protocols used in the domain, and the list of nodes that serve external requests.
+
+The domain's *membership* is the list of nodes
+that are allowed to participate in the network protocols inside the domain.
+A domain may be instantiated as one or more overlay networks that may or may not be inter-connected,
+depending on the protocols run in the domain.
+This allows both single-instance domains,
+as well as [grassroots](https://arxiv.org/abs/2301.04391) domains
```

Comment:
Maybe a brief compare contrast b/w grassroots domains and single domains beyond what is stated would be helpful